x-pack/plugins/security_solution/public/common/components/data_table/helpers.tsx

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { isEmpty } from 'lodash/fp';
+
+import type { EuiDataGridCellValueElementProps } from '@elastic/eui';
+import type { EuiTheme } from '@kbn/kibana-react-plugin/common';
+import type { SortColumnTable } from '../../../../common/types';
+import type { Ecs } from '../../../../common/ecs';
+import type { TimelineItem, TimelineNonEcsData } from '../../../../common/search_strategy';
+import type { ColumnHeaderOptions, SortDirection } from '../../../../common/types/timeline';
+
+/**
+ * Creates mapping of eventID -> fieldData for given fieldsToKeep. Used to store additional field
+ * data necessary for custom timeline actions in conjunction with selection state
+ * @param data
+ * @param eventIds
+ * @param fieldsToKeep
+ */
+export const getEventIdToDataMapping = (
+  timelineData: TimelineItem[],
+  eventIds: string[],
+  fieldsToKeep: string[],
+  hasAlertsCrud: boolean
+): Record<string, TimelineNonEcsData[]> =>
+  timelineData.reduce((acc, v) => {
+    const fvm =
```

Comment:
I don't think you wrote this, but we should rename explicitly to `fieldValueMapping` if possible.